browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -246,6 +251,17 @@ void OnBeforeURLRequestAdBlockTP(const ResponseCallback& next_callback,
           brave_shields::features::kBraveAdblockCnameUncloaking) &&
       !ctx->browser_context->IsTor();
 
+  // Also, skip CNAME uncloaking if there is currently a configured proxy.
+  Profile* profile = Profile::FromBrowserContext(ctx->browser_context);
+  ProxyConfigDictionary dict(
```

Comment:
can we use `ProxyServiceFactory::CreatePrefProxyConfigTrackerOfProfile` and then use `ProxyServiceFactory::CreateProxyConfigService` with the tracker to get `ProxyConfigService`?  So we can use `ProxyConfigService::GetLatestProxyConfig` to check the config availability is ProxyConfigService::ConfigAvailability::CONFIG_VALID` or not.

Reply:
Even better, with this approach we can check if it's configured as `ProxyConfig::ProxyRules::Type::PROXY_LIST`, which is equivalent to `SingleProxy` mode. Pushed this version.